### PR TITLE
Part of #17670: Replace Typography with Text component in RecoveryPhraseReminder

### PR DESCRIPTION
**Pull request by kremalicious**
Part of #17670.

Switch `Typography` components with `Text` component within the `RecoveryPhraseReminder `. Additionally, created story for it. And switched some markup around so that all items in that popover use the `Text` component, so all line height and text treatment is now consistent among all items.

### Before

Screenshot 2023-04-18 at 14 47 48

Screenshot 2023-04-18 at 14 47 54

### After

Screenshot 2023-04-18 at 14 53 54

Screenshot 2023-04-18 at 14 53 58

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/app/recovery-phrase-reminder/recovery-phrase-reminder.js [hunk 1]
```diff
@@ -35,27 +35,25 @@ export default function RecoveryPhraseReminder({ onConfirm, hasBackedUp }) {
...
-            <li>
+            <Text
+              as="li"
+              color={TextColor.textDefault}
+              fontWeight={FONT_WEIGHT.BOLD}
+            >
+              {t('recoveryPhraseReminderItemOne')}
+            </Text>
+            <Text as="li">{t('recoveryPhraseReminderItemTwo')}</Text>
+            <Text as="li">
```
Comment (DDDDDanica) [inline ui/components/app/recovery-phrase-reminder/recovery-phrase-reminder.js:56]:
```suggestion
           <li>
              <Text
                as="span"
                color={TextColor.textDefault}
                fontWeight={FontWeight.bold}
              >
                {t('recoveryPhraseReminderItemOne')}
              </Text>
            </li>
            <li>{t('recoveryPhraseReminderItemTwo')}</li>
            <li>
```

@georgewrmarshall Maybe i'm wrong here, it seems like we should keep the original structure, not sure if Text component will compile in the end. Can you double check this one?

File: ui/components/app/recovery-phrase-reminder/recovery-phrase-reminder.js [hunk 2]
```diff
@@ -35,27 +35,25 @@ export default function RecoveryPhraseReminder({ onConfirm, hasBackedUp }) {
...
-              >
-                {t('recoveryPhraseReminderItemOne')}
-              </Typography>
-            </li>
-            <li>{t('recoveryPhraseReminderItemTwo')}</li>
-            <li>
+            <Text
+              as="li"
+              color={TextColor.textDefault}
+              fontWeight={FONT_WEIGHT.BOLD}
```
Comment (DDDDDanica) [inline ui/components/app/recovery-phrase-reminder/recovery-phrase-reminder.js:51]:
```suggestion
              fontWeight={FontWeight.Bold}
```

File: ui/components/app/recovery-phrase-reminder/recovery-phrase-reminder.js (hunk 1 above)
Comment (DDDDDanica) [reply ui/components/app/recovery-phrase-reminder/recovery-phrase-reminder.js]: Makes sense, thanks for explaining, let's follow that !

**Review by DDDDDanica — APPROVED** [review]

**Review by brad-decker — APPROVED** [review]

**Comment by brad-decker** [comment]
Thanks for the contribution @kremalicious -- I'm updating your branch and seeing if we can get the build process passing, if it doesn't pass i'll investigate the issue and report here.

**Review by pedronfigueiredo — APPROVED** [review]

**Pull request merged**